stubs/seaborn/seaborn/regression.pyi

```diff
@@ -55,27 +58,61 @@ def lmplot(
     line_kws: dict[str, Any] | None = None,
     facet_kws: dict[str, Any] | None = None,
 ) -> FacetGrid: ...
+@overload
 def regplot(
-    data: pd.DataFrame | None = None,
+    data: None = None,
     *,
-    x: Incomplete | None = None,
-    y: Incomplete | None = None,
-    x_estimator: Incomplete | None = None,
-    x_bins: Incomplete | None = None,
+    x: _Vector | None = None,
+    y: _Vector | None = None,
+    x_estimator: Callable[[Incomplete], Incomplete] | None = None,
+    x_bins: int | _Vector | None = None,
     x_ci: Literal["ci", "sd"] | int | None = "ci",
     scatter: bool = True,
     fit_reg: bool = True,
     ci: int | None = 95,
     n_boot: int = 1000,
-    units: str | None = None,
+    units: _Vector | None = None,
     seed: _Seed | None = None,
     order: int = 1,
     logistic: bool = False,
     lowess: bool = False,
     robust: bool = False,
     logx: bool = False,
-    x_partial: str | None = None,
-    y_partial: str | None = None,
+    x_partial: _Vector | None = None,
+    y_partial: _Vector | None = None,
+    truncate: bool = True,
+    dropna: bool = True,
+    x_jitter: float | None = None,
+    y_jitter: float | None = None,
+    label: str | None = None,
+    color: ColorType | None = None,
+    marker: str = "o",
+    scatter_kws: dict[str, Any] | None = None,
+    line_kws: dict[str, Any] | None = None,
+    ax: Axes | None = None,
+) -> Axes: ...
+@overload
+def regplot(
+    data: pd.DataFrame,
+    *,
+    x: str | _Vector | None = None,
+    y: str | _Vector | None = None,
```

Comment:
`x`, `y` and the like can only be `str` if `data` is provided as a data frame, otherwise they must be vectors: https://seaborn.pydata.org/generated/seaborn.regplot.html